js/main.js

```diff
@@ -876,28 +884,57 @@ function createAttributeBalance(rowElement, effectTypes) {
                 }
             }
         }
-
-        for (const key in battles) {
-            /** @type {Battle} */
-            const battle = battles[key];
-            createAttributeBalanceEntry(
-                balanceElement,
-                battle.getReward.bind(battle),
-                () => battle.rewards,
-                effectType,
-                'Defeated ' + battle.title,
-                battle.isDone.bind(battle),
-            );
+        if (effectType === EffectType.MilitaryFactor) {
+            // let battlesWonLength = () => (Object.keys(battles).filter(key => battles[key].isDone() && battles[key].getReward(effectType)).length); not an option because getReward(effectType) returns 1 even if its +Military because it backs up to effectType.defaultValue of xMilitary which is 1
+            let battlesWonLength = () => {
+                let count = 0;
+                for (const key in battles) {
+                    const b = battles[key];
+                    if (b.isDone() && b.getReward(effectType)) {
```

Comment:
```suggestion
                    if (!b.isDone()) continue;
```
No need to go here through `getReward`, as you are already checking the right reward type in the coming lines. This just iterates all the rewards twice.

Also... nice chance for an early return (or early continue here :) ) - please adjust the rest of the if-block

Reply:
Ich finde die Idee mit "continue" super